### DISCUSSION TITLE
Angular slickgrid build failing with error - static ɵcmp: i0.ɵɵComponentDeclaration<AngularSlickgridComponent, "angular-slickgrid"

### DISCUSSION THREAD

**Discussion by meta-shubham-joshi** [ghiscoding/Angular-Slickgrid#987] [Q&A]
Angular slickgrid build(cmd: ng build project_name) failing with angular version 13.3.0, angular-slickgrid version 4.1.4, rxjs version 7.5.5. Attaching below the screenshot for your reference. 

![Screenshot (225)](https://user-images.githubusercontent.com/41181790/184885307-11cf8864-b568-4492-9aff-e2a927d8c425.png)

**Comment by ghiscoding** [comment]
I'm afraid that the Angular errors are too cryptic to provide any help here, the only Stack Overflow similar to your issue is this one [error TS2314: Generic type 'ɵɵComponentDeclaration' requires 7 type argument(s)](https://stackoverflow.com/questions/72701446/error-ts2314-generic-type-%c9%b5%c9%b5componentdeclaration-requires-7-type-arguments) which was answered by saying that 1 of the lib used required Angular 14 and that might match your issue but I can't really tell since you didn't include the entire error (comparing to the SO question which mentioned Angular 14). That seems to be more of an Angular error, not really an Angular-Slickgrid error

Another suggestion would be to make sure that your Angular-Slickgrid version matches the Slickgrid-Universal versions that were originally assigned to them, every time I update Slickgrid-Universal, I then have to release a new version of Angular-Slickgrid because they have to follow each other. If your Angular-Slickgrid is on `^4.1.4` then **all** your Slickgrid-Universal (if you have any) must be on `^1.3.0` as shown in link below (the portion on the left in red)

https://github.com/ghiscoding/angular-slickgrid-demos/commit/8073d2fef98393b7343abe40e680d7eaeca63749

It might be preferable to update to latest versions if you can, also note that the last version (few days ago) has a small 1 line breaking change, so make sure you do that 1 line change whenever you update to latest
https://github.com/ghiscoding/Angular-Slickgrid/releases

**Comment by meta-shubham-joshi** [comment]
Actually that was related to carrot icon (^) before the version no in
package json file. So, it's resolved now. Thanks for the immediate reply.

**Reply by ghiscoding** [reply]
decided to downgrade back to Angular 13 and publish [v4.3.1](https://github.com/ghiscoding/Angular-Slickgrid/releases/tag/v4.3.1), so you should be able to update to the latest version now. However make sure that all your Slickgrid-Universal are also up to date. Cheers and thanks for using the lib.
